lisa/sut_orchestrator/hyperv/platform_.py

```diff
@@ -0,0 +1,295 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import random
+import string
+from pathlib import PurePosixPath, PureWindowsPath
+from typing import Any, List, Type
+
+from lisa import feature, schema, search_space
+from lisa.environment import Environment
+from lisa.node import RemoteNode
+from lisa.platform_ import Platform
+from lisa.tools import Cp, HyperV, Ls, PowerShell, Unzip
+from lisa.util.logger import Logger, get_logger
+
+from .. import HYPERV
+from .context import NodeContext, get_node_context
+from .schema import HypervNodeSchema, HypervPlatformSchema
+from .serial_console import SerialConsole, SerialConsoleLogger
+
+
+class _HostCapabilities:
+    def __init__(self) -> None:
+        self.core_count = 0
+        self.free_memory_kib = 0
+
+
+class HypervPlatform(Platform):
+    def __init__(
+        self,
+        runbook: schema.Platform,
+    ) -> None:
+        super().__init__(runbook=runbook)
+
+    @classmethod
+    def type_name(cls) -> str:
+        return HYPERV
+
+    @classmethod
+    def supported_features(cls) -> List[Type[feature.Feature]]:
+        return [SerialConsole]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        hyperv_runbook = self.runbook.get_extended_runbook(HypervPlatformSchema)
+        assert hyperv_runbook, "platform runbook cannot be empty"
+        self._hyperv_runbook = hyperv_runbook
+        print("Hyperv platform initialize")
+        print(f"{self._hyperv_runbook}")
+
+        if len(self._hyperv_runbook.servers) > 1:
+            self._log.warning(
```

Comment:
info level is enough.

Reply:
Isn't warning more proper here since we are _warning_ the user that the way they have configured the runbook is not supported? Also, Libvirt platform uses warning for this.

Reply:
Warning is not a clear expectation, because it highly depends on if user see it or not. Either raise exception to stop run, or use info to log it.

Reply:
changed.